pkg/storage/edv/restprovider.go

```diff
@@ -206,6 +208,38 @@ func (r *RESTProvider) Close() error {
 	return nil
 }
 
+func (r *RESTProvider) AddEncryptedIndices(k string, encryptedDocumentBytes []byte,
+	restStore *restStore) (*models.EncryptedDocument, error) {
+	var encryptedDocument models.EncryptedDocument
+
+	err := json.Unmarshal(encryptedDocumentBytes, &encryptedDocument)
+	if err != nil {
+		return nil, fmt.Errorf(failUnmarshalValueIntoEncryptedDocument, err)
+	}
+
+	indexedAttributeCollection, err := restStore.createIndexedAttributes(k)
+	if err != nil {
+		return nil, fmt.Errorf(failCreateIndexedAttributes, err)
+	}
+
+	encryptedDocument.IndexedAttributeCollections = indexedAttributeCollection
+
+	return &encryptedDocument, nil
+}
+
+func (r *RESTProvider) RetrieveEDVDocumentID(k string, restStore *restStore) (string, error) {
+	return restStore.retrieveEDVDocumentID(k)
+}
+
+func (r *RESTProvider) Batch(batch *models.Batch) error {
```

Comment:
Should there be a return value that contains the response of each update, e.g. CouchDB returns the following information for each item in the batch:

```
	ID     string
	Error  string
	Reason string
	Ok     bool
	Rev    string
```

Reply:
The EDV server will return an array of responses, one for each item in the batch. Each individual response follows the spec for the [Create Document, Update Document and Delete Document endpoints](https://identity.foundation/confidential-storage/#http-api). In the case of an error, then the response will contain the specific reason. The `error` returned from this RESTProvider Batch method will contain all of this information... however, all Aries cares about is whether there it succeeded or failed... if it failed, then the full error will get printed out.

The lower level restClient will actually return the array of responses in case someone wants to look at them. (But for the Aries provider, we have no use for this information)

Reply:
(Note: the full error will contain the responses for each item in the batch, so that way we can figure out which item failed.)